test/extended/networking/livemigration.go

```diff
@@ -0,0 +1,463 @@
+	nadclient "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/clientset/versioned/typed/k8s.cni.cncf.io/v1"
+
+	"github.com/openshift/origin/test/extended/networking/kubevirt"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = Describe("[sig-network][OCPFeatureGate:PersistentIPsForVirtualization][Feature:Layer2LiveMigration] Kubevirt Virtual Machines", func() {
+	oc := exutil.NewCLIWithPodSecurityLevel("network-segmentation-e2e", admissionapi.LevelBaseline)
+	f := oc.KubeFramework()
+
+	InOVNKubernetesContext(func() {
+		var (
+			cs         clientset.Interface
+			nadClient  nadclient.K8sCniCncfIoV1Interface
+			virtClient *kubevirt.Client
+		)
+
+		BeforeEach(func() {
+			cs = f.ClientSet
+
+			var err error
+			nadClient, err = nadclient.NewForConfig(f.ClientConfig())
+			Expect(err).NotTo(HaveOccurred())
+			virtClient, err = kubevirt.NewClient(oc, "/tmp/")
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		Context("with user defined networks and persistent ips configured", func() {
+			const (
+				nadName           = "blue"
+				bindingName       = "passt"
+				udnCrReadyTimeout = 5 * time.Second
+				vmName            = "myvm"
+			)
+			var (
+				cidrIPv4 = "203.203.0.0/16"
+				cidrIPv6 = "2014:100:200::0/60"
+			)
+
+			DescribeTableSubtree("created using",
+				func(createNetworkFn func(netConfig networkAttachmentConfigParams)) {
+
+					DescribeTable("[Suite:openshift/network/virtualization] should keep ip", func(netConfig networkAttachmentConfigParams, vmResource string, opCmd func(cli *kubevirt.Client, vmNamespace, vmName string)) {
+						var err error
+						netConfig.namespace = f.Namespace.Name
+						// correctCIDRFamily makes use of the ginkgo framework so it needs to be in the testcase
+						netConfig.cidr = correctCIDRFamily(oc, cidrIPv4, cidrIPv6)
+						workerNodes, err := getWorkerNodesOrdered(cs)
+						Expect(err).NotTo(HaveOccurred())
+						Expect(len(workerNodes)).To(BeNumerically(">=", 2))
+
+						isDualStack := getIPFamilyForCluster(f) == DualStack
+
+						createNetworkFn(netConfig)
+
+						httpServerPods := prepareHTTPServerPods(f, netConfig, workerNodes)
+						vmCreationParams := kubevirt.CreationTemplateParams{
+							VMName:                    vmName,
+							VMNamespace:               f.Namespace.Name,
+							FedoraContainterDiskImage: "quay.io/kubevirt/fedora-with-test-tooling-container-disk:20241024_891122a6fc",
+						}
+						if netConfig.role == "primary" {
+							vmCreationParams.NetBindingName = bindingName
+						} else {
+							vmCreationParams.NetworkName = nadName
+						}
+
+						Expect(virtClient.CreateVM(vmResource, vmCreationParams)).To(Succeed())
+						waitForVMReadiness(virtClient, vmCreationParams.VMNamespace, vmCreationParams.VMName)
+
+						By("Retrieving addresses before test operation")
+						expectedAddresses := obtainAddresses(virtClient, netConfig, vmName)
```

Comment:
```suggestion
						initialAddresses := obtainAddresses(virtClient, netConfig, vmName)
```

or `addressesBeforeMigration`